lib/ansible/plugins/connection/paramiko_ssh.py

```diff
@@ -399,53 +409,108 @@ def exec_command(self, cmd, in_data=None, sudoable=True):
 
         cmd = to_bytes(cmd, errors='surrogate_or_strict')
 
-        no_prompt_out = b''
-        no_prompt_err = b''
-        become_output = b''
+        event = threading.Event()
+        chan.in_buffer.set_event(event)
+        chan.in_stderr_buffer.set_event(event)
 
-        try:
-            chan.exec_command(cmd)
-            if self.become and self.become.expect_prompt():
-                passprompt = False
-                become_sucess = False
-                while not (become_sucess or passprompt):
-                    display.debug('Waiting for Privilege Escalation input')
+        b_stdout = b_stderr = b''
+        b_tmp_stdout = b_tmp_stderr = b''
 
+        chan.exec_command(cmd)
+        if self.become and self.become.expect_prompt():
+            while not (self._flags['become_success'] or chan.exit_status_ready()):
```

Comment:
I'd put `(self._flags['become_success'] or chan.exit_status_ready())` into a `@property` for readability so that it'd look like
```suggestion
            while not self._cmd_finished:
```